apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/env/config/UserResourceLocator.java

```diff
@@ -1,81 +1,81 @@
-            searchDirs.add(dirProvider.getUserConfigLocation() + File.separator + dirProvider.getConfigName());
-        }
-        if (log.isTraceEnabled()) {
-            log.trace("The directory list:");
-            searchDirs.forEach(s -> log.trace("  {}", s));
-        }
-        FileInputStream is = null;
-        for (String dir : searchDirs) {
-            String p = dir + File.separator + resourceName;
-            try {
-                is = new FileInputStream(p);
-                log.info("Located in directories, resource={}", p);
-                break;
-            } catch (FileNotFoundException ignored) {
-                log.info("File not found: " + p); // do not use logger (we should init it before using)
-            }
-        }
-        return Optional.ofNullable(is);
-    }
-}
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.util.env.config;
+
+import com.apollocurrency.aplwallet.apl.util.StringUtils;
+import com.apollocurrency.aplwallet.apl.util.env.dirprovider.ConfigDirProvider;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Slf4j
+public class UserResourceLocator implements ResourceLocator {
+
+    private final ConfigDirProvider dirProvider;
+    private final String configDir;
+
+    @Inject
+    public UserResourceLocator(ConfigDirProvider dirProvider) {
+        this.dirProvider = dirProvider;
+        this.configDir = dirProvider.getConfigName();
+    }
+
+    @Override
+    public Optional<InputStream> locate(final String resourceName) {
+        Objects.requireNonNull(resourceName);
+        return locateInDirectories(resourceName).or(() -> locateInResources(resourceName));
+    }
+
+    private Optional<InputStream> locateInResources(String resourceName) {
+        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+        //Windows uses path with backslashes but in java resources it should be simple slash
+        Path p = Path.of(dirProvider.getConfigName(), resourceName);
+        String path = p.normalize().toString().replaceAll("\\\\", "/");
+        InputStream is = classloader.getResourceAsStream(path);
+        if (is != null) {
+            log.info("Located in resources, resource={}", path);
+        }else{
+            log.warn("Can not find resource at: {}",path);
+        }
+
+        return Optional.ofNullable(is);
+    }
+
+    private Optional<InputStream> locateInDirectories(String resourceName) {
+        List<String> searchDirs = new ArrayList<>();
+        if (!StringUtils.isBlank(configDir)) { //load just from confDir
+            searchDirs.add(configDir);
+        }// else { //go trough standard search order and load all
```

Comment:
Why did you commented this out? All i knew about this code, that it should represent real node configuration search order

Reply:
I guess the user config path should always be in the path list to search resources.
We should use UserResourceLocator to load resources in all places of the application.